apps/music/test/marionette/Player_test.js

```diff
@@ -185,4 +185,23 @@ marionette('Music player tests', function() {
     });
   });
 
+  suite('Player navigation', function() {
+    test('Check back button works', function() {
+      music.launch();
+      music.waitForFirstTile();
+      music.switchToSongsView();
+
+      var title = music.header.findElement('#title-text').text();
+
+      music.playFirstSong();
+
+      assert.notEqual(title, music.header.findElement('#title-text').text());
+
+      music.tapHeaderActionButton();
+
```

Comment:
Fixed it in the current update of the PR.
Calling ````music.firstSong;```` ought to be enough: it waits on the visibility.